sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -357,8 +357,15 @@ ESIMD_INLINE
     __esimd_scatter_scaled<PromoT, N, decltype(si), TypeSizeLog2, scale>(
         mask.data(), si, glob_offset, offsets.data(), promo_vals.data());
   } else {
-    __esimd_scatter_scaled<T, N, decltype(si), TypeSizeLog2, scale>(
-        mask.data(), si, glob_offset, offsets.data(), vals.data());
+    using Treal = __raw_t<T>;
+    if constexpr (!std::is_same_v<Treal, T>) {
+      simd<Treal, N> Values = convert<Treal>(vals);
```

Comment:
I think this is still wrong. This `convert` is type conversion, which will change bits of the value and incorrect value will be written. E.g. T=tf32 and Treal=int. Only bitcasts (sycl::bit_cast) should be used.
Are tests passing with this?